src/main/kotlin/org/misarch/discount/graphql/model/connection/DiscountUsageConnection.kt

```diff
@@ -0,0 +1,69 @@
+package org.misarch.discount.graphql.model.connection
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.federation.directives.ShareableDirective
+import com.querydsl.core.types.Expression
+import com.querydsl.core.types.dsl.BooleanExpression
+import com.querydsl.core.types.dsl.ComparableExpression
+import com.querydsl.sql.SQLQuery
+import org.misarch.discount.graphql.AuthorizedUser
+import org.misarch.discount.graphql.model.DiscountUsage
+import org.misarch.discount.graphql.model.connection.base.*
+import org.misarch.discount.persistence.model.DiscountUsageEntity
+import org.misarch.discount.persistence.repository.DiscountUsageRepository
+
+/**
+ * A GraphQL connection for [DiscountUsage]s.
+ *
+ * @param first The maximum number of items to return
+ * @param skip The number of items to skip
+ * @param predicate The predicate to filter the items by
+ * @param order The order to sort the items by
+ * @param repository The repository to fetch the items from
+ * @param authorizedUser The authorized user
+ * @param applyJoin A function to apply a join to the query
+ */
+@GraphQLDescription("A connection to a list of `DiscountUsage` values.")
+@ShareableDirective
+class DiscountUsageConnection(
+    first: Int?,
+    skip: Int?,
+    predicate: BooleanExpression?,
+    order: DiscountUsageOrder?,
+    repository: DiscountUsageRepository,
+    authorizedUser: AuthorizedUser?,
+    applyJoin: (query: SQLQuery<*>) -> SQLQuery<*> = { it }
+) : BaseConnection<DiscountUsage, DiscountUsageEntity>(
+    first,
+    skip,
+    null,
+    predicate,
+    (order ?: DiscountUsageOrder.DEFAULT).toOrderSpecifier(DiscountUsageOrderField.ID),
+    repository,
+    DiscountUsageEntity.ENTITY,
+    authorizedUser,
+    applyJoin
+) {
+
+    override val primaryKey: ComparableExpression<*> get() = DiscountUsageEntity.ENTITY.id
+
+}
+
+@GraphQLDescription("DiscountUsage order fields")
+enum class DiscountUsageOrderField(override vararg val expressions: Expression<out Comparable<*>>) : BaseOrderField {
+    @GraphQLDescription("Order discount usages by their id")
+    ID(DiscountUsageEntity.ENTITY.id),
+
+    @GraphQLDescription("Order discount usages by their usages")
+    USAGES(DiscountUsageEntity.ENTITY.usages, DiscountUsageEntity.ENTITY.id),
```

Comment:
What about `order by expiration date` and `order by creation date`?

Reply:
ohm, since when does a DiscountUsage has either of those?
a discount usage just keeps track how often a discount has been used (to calculate how often it can still be used)
do we need more tracking then?

Reply:
I don't have a creation date
but I will add the valid before / after things

Reply:
added with c153d9f043045901a542973f40a59a03687d326a